File: DataServer/gRpcServer/src/main/java/via/sep3/grpcserver/Entities/Car.java
package via.sep3.grpcserver.Entities;

import org.checkerframework.checker.signature.qual.Identifier;

import javax.persistence.*;

@Entity
@Table(name = "car",schema = "trip_share")
public class Car {
    @Id
    @Column
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column(nullable = false,unique = true)
    private String plateNumber;


    @OneToOne
    @JoinColumn(name = "fk_drive_id",unique = true,nullable = false,referencedColumnName = "email")
    private User driver;
    @Column(nullable = false)
    private String color;
    @Column(nullable = false)
    private int seatsCount;
    @Column(nullable = false)
    private String fuelType;
    @Column
    private String model;


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getDriver() {
        return driver;
    }

    public void setDriver(User driver) {
        this.driver = driver;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public int getSeatsCount() {
        return seatsCount;
    }

    public void setSeatsCount(int seatsCount) {
        this.seatsCount = seatsCount;
    }

    public String getFuelType() {
        return fuelType;
    }

    public void setFuelType(String fuelType) {
        this.fuelType = fuelType;
    }

    public String getPlateNumber() {
        return plateNumber;
    }

    public void setPlateNumber(String plateNumber) {
        this.plateNumber = plateNumber;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }
}

File: LogicServer/Domain/Model/Car.cs

using System.Net;
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Domain.Model;

public class Car
{
    [JsonPropertyName("registration_number")]
    public string PlateNumber { get; set; }
    [JsonPropertyName("color")]
    public string Color { get; set; }
    [JsonPropertyName("model")]
    public string CarModel { get; set; }
    [JsonPropertyName("seats")]
    public int? SeatsCount { get; set; }
    [JsonPropertyName("fuel_type")]
    public string FuelType { get; set; }


    public Car( string plateNumber, string color, string carModel, int? seatsCount, string fuelType)
    {
        PlateNumber = plateNumber;
        Color = color;
        CarModel = carModel;
        SeatsCount = seatsCount;
        FuelType = fuelType;
    }
    
}
    